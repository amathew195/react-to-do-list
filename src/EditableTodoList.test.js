import React from "react";
import { render } from "@testing-library/react";
import EditableTodoList from "./EditableTodoList";

const top1 = {
  id: 1,
  title: 'top1',
  priority: 1,
  description: 'mock1',
}

const top2 = {
  id: 2,
  title: 'top2',
  priority: 2,
  description: 'mock2',
}

const top3 = {
  id: 3,
  title: 'top3',
  priority: 3,
  description: 'mock3',
}

const todos = [top1, top2, top3];

describe("renders correctly", function () {
  it("renders", function () {
    render(<EditableTodoList todos={todos}/>);
  })
})

describe("snapshots", function() {
  it("matches snapshot", function() {
      const { asFragment } = render(<EditableTodoList todos={todos}/>);
      expect(asFragment()).toMatchSnapshot();
  });
});

describe("displays appropriate number of todos", function(){
  it("displays 0 todos when no todos", function() {
    const { container, debug } = render(<EditableTodoList todos={[]}/>);

    expect(container.querySelectorAll(".EditableTodo").length).toEqual(0);
  })

  it("displays appropriate amount of todos", function() {
    const { container, debug } = render(<EditableTodoList todos={todos}/>);

    expect(container.querySelectorAll(".EditableTodo").length).toEqual(3);
  })
})
