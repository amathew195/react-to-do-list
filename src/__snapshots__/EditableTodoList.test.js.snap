// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="EditableTodoList"
  >
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              top1
            </b>
             
            <small>
              (priority: 1)
            </small>
          </div>
          <div>
            <small>
              mock1
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              top2
            </b>
             
            <small>
              (priority: 2)
            </small>
          </div>
          <div>
            <small>
              mock2
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              top3
            </b>
             
            <small>
              (priority: 3)
            </small>
          </div>
          <div>
            <small>
              mock3
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
